cfs-require-import.php

```diff
@@ -8,32 +8,50 @@
 Author URI: https://www.vanpattenmedia.com/
 */
 
-class CfsRequireImport {
+include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
 
+class CfsRequireImport {
+	
 	function __construct() {
-		register_activation_hook( plugin_basename( __FILE__ ), array( $this, 'import_cfs_fields' ) );
+		
+		/* Register activation hook. */
+		register_activation_hook( plugin_basename( __FILE__ ), array( $this , 'import_cfs_fields' ) );
+		
+		/* Add admin notice */
+		add_action( 'admin_notices' , array( $this , 'print_import_notice' ) );
 	}
-
+	
+ 	/* Runs only when the plugin is activated. */
 	public static function import_cfs_fields() {
 		$fields = file_get_contents( trailingslashit( dirname( __FILE__ ) ) . 'fields.json' );
 
 		$options = array(
-			'import_code' => json_decode( stripslashes( $fields ), true ),
+			'import_code' => json_decode( $fields , true ),
 		);
 
 		$result = CFS()->field_group->import( $options );
-
-		// FIXME
-		add_action( 'admin_notices', function() {
-			echo '<div class="updated"><p>' . strip_tags( $result ) . '</p></div>';
-		} );
+		
+		/* Create transient data */
+		set_transient( 'cfs_import_result' , $result , 60 );
+	}
+	
+	/* Admin Notice on Activation. */
+	public static function print_import_notice() {
+		
+		/* Check transient, if available display notice */
+		if ( get_transient( 'cfs_import_result' ) ) {
```

Comment:
Just so I understand, is the transient used to persist the admin notice across the page reload? Sort of a fake session handler?

Reply:
`admin_notices` don't work inside `register_activation_hook` and one way that i know is to solve this problem is to use transient data. When activation triggered - transient data setted with notice, then notice echoes 1 time and transient data deleted, therefore notice didn't appeares until next activation. Sorry, I don't know what is "fake session handler".